src/App.tsx

```diff
@@ -1,14 +1,106 @@
-/* eslint-disable max-len */
-import React from 'react';
+/* eslint-disable no-console */
+/* eslint-disable import/no-cycle */
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
+import { Todo } from './types/Todo';
+import { getTodos, getUser } from './api';
 import { Loader } from './components/Loader';
+import { TodoModal } from './components/TodoModal';
+import { User } from './types/User';
+
+export const SortFieldEnum = {
+  COMPLETED: 'completed',
+  ACTIVE: 'active',
+  NONE: '',
+};
+
+type FilterArgs = {
+  searchField: string;
+  sortField: (typeof SortFieldEnum)[keyof typeof SortFieldEnum];
+};
+
+function filteredTodos(todos: Todo[], { searchField, sortField }: FilterArgs) {
+  let visibleFilteredTodos = [...todos];
+
+  if (searchField) {
+    const searchFieldNormalize = searchField.toLowerCase().trim();
+
+    visibleFilteredTodos = visibleFilteredTodos.filter((todo) => {
+      return todo.title.toLowerCase().trim().includes(searchFieldNormalize);
+    });
+  }
+
+  if (sortField) {
+    visibleFilteredTodos = visibleFilteredTodos.filter((todo) => {
+      switch (sortField) {
+        case SortFieldEnum.ACTIVE:
+          return !todo.completed;
+        case SortFieldEnum.COMPLETED:
+          return todo.completed;
+        default:
+          return todo;
+      }
+    });
+  }
+
+  return visibleFilteredTodos;
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [visibleTodo, setVisibleTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [visibleUser, setVisibleUser] = useState<User | null>(null);
+  const [isLoadingUser, setIsLoadingUser] = useState(false);
+  const [searchField, setSearchField] = useState('');
+  const [sortField, setSortField]
+    = useState<(typeof SortFieldEnum)[keyof typeof SortFieldEnum]>('');
+
+  const getUserById = (userId: number) => {
+    setIsLoadingUser(true);
+    getUser(userId)
+      .then((user) => setVisibleUser(user))
+      .catch((error) => {
+        console.warn(error);
+      })
+      .finally(() => setIsLoadingUser(false));
+  };
```

Comment:
you can move all user-related logic to the TodoModal component, why do you need it directly in the App component?
You click on todo and set coressponding todo object to state -> show the modal and take the user ID from the selected todo object and make request

Reply:
Hi) cause I create the function getUserById, this function use use this states (  const [visibleUser, setVisibleUser] = useState<User | null>(null);
  const [isLoadingUser, setIsLoadingUser] = useState(false);) and then I use this function as a props in TodoList.